app/routers/reset_password.py

```diff
@@ -0,0 +1,116 @@
+from typing import Optional
+
+from fastapi import APIRouter, Depends, Request
+from pydantic import ValidationError
+from sqlalchemy.orm import Session
+from starlette.responses import RedirectResponse
+from starlette.status import HTTP_302_FOUND
+
+from app.database.models import User
+from app.dependencies import get_db, templates
+from app.internal.security.ouath2 import (
+    authenticate_user, check_jwt_token,
+    create_jwt_token, update_password)
+from app.internal.security.reset_password import (
+    BackgroundTasks, send_mail)
+from app.internal.security.schema import (
+    ForgotPassword, ResetPassword)
+
+
+router = APIRouter(
+    prefix="",
+    tags=["/reset_password"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+@router.get("/forgot-password")
+async def forgot_password_form(
+            request: Request) -> templates:
+    """rendering forgot password form get method"""
+    return templates.TemplateResponse("forgot_password.html", {
+        "request": request,
+    })
+
+
+@router.post('/forgot-password')
+async def forgot_password(
+        request: Request, background_tasks: BackgroundTasks,
+        db: Session = Depends(get_db)) -> templates:
+    """
+    Validaiting form data fields.
+    Validaiting form data against database records.
+    If all validations succeed, creating jwt token,
+    then sending email to the user with a reset password route link.
+    The contains the verafiction jwt token.
+    """
+    form = await request.form()
+    form_dict = dict(form)
+    try:
+        # validating form data by creating pydantic schema object
+        user = ForgotPassword(**form_dict)
+    except ValidationError:
+        user = False
+    if user:
+        user = await authenticate_user(db, user, email=True)
+        if user:
+            user.token = create_jwt_token(user, jwt_min_exp=15)
+            await send_mail(db, user, background_tasks)
+            return templates.TemplateResponse("forgot_password.html", {
+                "request": request,
+                "message": "Email for reseting password was sent"})
+    return templates.TemplateResponse("forgot_password.html", {
+            "request": request,
+            "message": 'Please check your credentials'
+        })
+
+
+@router.get("/reset-password")
+async def reset_password_form(
+        request: Request, token: Optional[str] = ""
+        ) -> templates:
```

Comment:
why not RedirectResponse?

Reply:
This function handles 'GET' method request, and renders the empty html form.